### PR TITLE
feat: initial mediator

### PR DESCRIPTION
**Pull request by bcornett**

### CODE REVIEW

File: Turner.Infrastructure.Mediator/BackgroundJobs/BackgoundJobExecutor.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            _mediator = mediator;
+        }
+
+        public async Task Execute(T command)
+        {
+            var result = await _mediator.HandleAsync(command);
+            if (result.HasErrors)
+            {
+                throw new Exception(JsonConvert.SerializeObject(result.Errors));
```
Comment (justinobney) [inline Turner.Infrastructure.Mediator/BackgroundJobs/BackgoundJobExecutor.cs:21]: Nice.

File: Turner.Infrastructure.Mediator/Decorators/TransactionHandler.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    public class TransactionHandler<TRequest> : IRequestHandler<TRequest> where TRequest : IRequest
+    {
+        private readonly Func<IRequestHandler<TRequest>> _decorateeFactory;
+        private readonly TransactionBaseHandler<TRequest, Response> _transactionHandler;
+
+        public TransactionHandler(Func<IRequestHandler<TRequest>> decorateeFactory,
+            TransactionBaseHandler<TRequest, Response> transactionHandler)
+        {
+            _decorateeFactory = decorateeFactory;
```
Comment (justinobney) [inline Turner.Infrastructure.Mediator/Decorators/TransactionHandler.cs:48]: what is this 0_o

File: Turner.Infrastructure.Mediator/Error.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+namespace Turner.Infrastructure.Mediator
+{
+    public class Error
+    {
+        public string PropertyName { get; set; }
```
Comment (justinobney) [inline Turner.Infrastructure.Mediator/Error.cs:5]: I have always been up in the air about this. I guess it is assumed that you leave the PropertyName blank if it cant be tied to a single property?

**Review by justinobney — COMMENTED** [review]
Looks good.

File: Turner.Infrastructure.Mediator/Error.cs (hunk 3 above)
Comment (bcornett) [reply Turner.Infrastructure.Mediator/Error.cs:5]: @justinobney Yeah, that's been how we've always done it...

File: Turner.Infrastructure.Mediator/Decorators/TransactionHandler.cs (hunk 2 above)
Comment (bcornett) [reply Turner.Infrastructure.Mediator/Decorators/TransactionHandler.cs:48]: http://simpleinjector.readthedocs.io/en/latest/aop.html#decoration

They use `decoratee`. What specifically doesn't make sense?

**Pull request merged**